.travis.yml

```diff
@@ -54,19 +56,37 @@ matrix:
       addons: {"apt":{"sources":["hvr-ghc"],"packages":["ghc-7.4.2","cabal-install-2.4","freeglut3-dev"]}}
     - compiler: ghc-head
       addons: {"apt":{"sources":["hvr-ghc"],"packages":["ghc-head","cabal-install-head","freeglut3-dev"]}}
-      env: GHCHEAD=true
   allow_failures:
     - compiler: ghc-head
     - compiler: ghc-8.8.1
 before_install:
+  - |
+    if [ "$TRAVIS_OS_NAME" = "linux" ]; then
+      sudo add-apt-repository -y ppa:hvr/ghcjs;
+      curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | sudo apt-key add -
+      sudo apt-add-repository 'https://deb.nodesource.com/node_8.x xenial main'
+      sudo apt-get update;
+      sudo apt-get install $CC cabal-install-3.0 nodejs freeglut3-dev;
```

Comment:
While installing `nodejs` with `apt-get install` appears to (unfortunately) be necessary, do we need to install these other things (`cabal-install-3.0`, `freeglut3-dev`, etc.) with `apt-get install` now that we're back on Xenial? Furthermore, do we need to install `nodejs` on non-GHCJS configurations?

Reply:
Good point, let me split that line.
Trying to `apt-get install` what's already installed is no-op, so installing `$CC freeglug3-dev` is cheap.

Reply:
done.